pkg/proof/proof_test.go

```diff
@@ -256,3 +259,21 @@ func TestAllSharesInclusionProof(t *testing.T) {
 	require.NoError(t, err)
 	assert.NoError(t, proof.Validate(dataRoot))
 }
+
+// Ensure that we reject negative index values and avoid overflows.
+// https://github.com/celestiaorg/celestia-app/issues/3140
+func TestQueryTxInclusionProofRejectsNegativeValues(t *testing.T) {
```

Comment:
On re-review, I failed to catch that this test function could've been added to a new file `querier_test.go` considering it's testing a function defined in `querier.go`.

Reply:
I initially had it in regression_test.go then go asked to move it here

Reply:
![image](https://github.com/celestiaorg/celestia-app/assets/4898263/9c43e19d-cbb2-4b68-8760-8907bcbae399)

Reply:
Yea IMO that comment is valid. It comes from Callum not Go. `regression_test.go` doesn't make sense to me because there is no source code file called `regression.go`. The convention followed in this code base is that test files suffix the source file with `_test.go`. For example:

- Implementation in: `querier.go`
- Tests in: `querier_test.go`